shell/app-shell/elements/shell-ui.js

```diff
@@ -210,6 +211,75 @@ class ShellUi extends Xen.Debug(Xen.Base, log) {
   _onShare(e, share) {
     this._fire('share', share);
   }
+  _onMicStart() {
+    this._setState({micHidden: false});
+  }
+  _onMicEnd(e) {
+    this._setState({micHidden: true});
+    const finalTranscript = e.detail.toLowerCase().replace(/[^A-Za-z0-9 ]/g, '');
```

Comment:
Small helper method to do the `toLowerCase().replace(...)` seems worthwhile since it's in three places so far.